### PR TITLE
Update Debugging and TypeScript documentation

### PR DESCRIPTION
**Pull request by pako88**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

I updated the Debugging and TypeScript Documentation.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by mgrybyk** [comment]
Can you please clarify how would the nightly debugger help? Didn't have any issues debugging typescript code with ts node without any plugins

**Comment by pako88** [comment]
debugging was broken for me for several months now.
actually debugging in top process worked (onprepare for example), but as soon as i wanted to set a breakpoint at test level, the breakpoint was just ignored.

today i found the time to look into this a little deeper and i tried many things.
ultimately i came to following ticket, in which the nightly version was highlighted: https://github.com/microsoft/vscode/issues/82523#issuecomment-609934308
i tried it and *boom*, it worked.
i don't know the technical answer to this.

**Comment by mgrybyk** [comment]
@pako88 you have to enable auto attach to the child process which is mentioned in the docs.

**Comment by pako88** [comment]
yes, i know. i had this enabled all the time

**Comment by mgrybyk** [comment]
That's strange, didn't get any complains on that. Just verified, worked without any additional plugins out of the box following https://webdriver.io/docs/debugging.html#debugging-with-visual-studio-code-vscode

**Comment by pako88** [comment]
it worked for me also a long time until beginning of this year. i think it brokes with a vscode update...
it definitively broked without changing anything to the launch configuration

**Comment by mgrybyk** [comment]
Can it be something with your local setup?

**Comment by pako88** [comment]
i don't think so - i tried many things and couldn't get it working until i installed the nightly version of the debugger.

which vscode version do you have? I'm with 1.46.1

**Comment by pako88** [comment]
additionally in the linked ticket they say that this bug doesn't appear with a mac, only with linux and windows.
so if you're using a mac, maybe this is the reason it works for you.

**Comment by christian-bromann** [comment]
> additionally in the linked ticket they say that this bug doesn't appear with a mac, only with linux and windows.

How about we add this sentence to the bottom of the section as a note `> ...` then?

**Comment by pako88** [comment]
i made the changes.
@christian-bromann and @mgrybyk can you review them?

File: docs/TypeScript.md [hunk 1]
```diff
@@ -107,6 +107,8 @@ const config: WebdriverIO.Config = {
 export { config }
 ```
 
+If you are using this approach for a typed configuration, you have to remove the line with 'ts-node/register' from your framework options in your config file.
```
Comment (mgrybyk) [inline docs/TypeScript.md:110]: Do we still need this one?

Comment (pako88) [reply docs/TypeScript.md:110]: Yes, if i don't remove this i have several ts compilation errors in combination with the preview of the javascript debugger

**Pull request merged**